openqabot/incrementapprover.py

```diff
@@ -0,0 +1,116 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: MIT
+from argparse import Namespace
+from logging import getLogger
+from pprint import pformat
+
+import osc.conf
+import osc.core
+
+from openqabot.openqa import openQAInterface
+
+from . import OBS_GROUP, OBS_URL
+
+log = getLogger("bot.increment_approver")
+
+
+class IncrementApprover:
+    def __init__(self, args: Namespace) -> None:
+        self.args = args
+        self.token = {"Authorization": "Token {}".format(args.token)}
+        self.client = openQAInterface(args)
+        osc.conf.get_config(override_apiurl=OBS_URL)
+
+    def __call__(self) -> int:
```

Comment:
this method is very long. Can you extract methods or helper functions?. For example whenever you used a comment like below "find the latest request on OBS", why not make that a function?